compute_tools/src/compute.rs

```diff
@@ -1385,3 +1391,87 @@ pub fn forward_termination_signal() {
         kill(pg_pid, Signal::SIGQUIT).ok();
     }
 }
+
+// Start a thread that will periodically renew LSN lease for static compute
+fn start_lsn_lease_for_static(compute_state: &ComputeState) {
+    let spec = compute_state.pspec.as_ref().expect("spec must be set");
+
+    let lsn = match spec.spec.mode {
+        ComputeMode::Static(lsn) => lsn,
+        _ => return,
+    };
+
+    let conn_strings = spec.pageserver_connstr.split(',');
+
+    let configs = conn_strings
+        .map(|connstr| {
+            let mut config = postgres::Config::from_str(connstr).expect("invalid connstr");
+            if let Some(storage_auth_token) = &spec.storage_auth_token {
+                info!("Got storage auth token from spec file");
+                config.password(storage_auth_token.clone());
+            } else {
+                info!("Storage auth token not set");
+            }
+            config
+        })
+        .collect::<Vec<_>>();
+
+    let cmd = format!("lease lsn {} {} {} ", spec.tenant_id, spec.timeline_id, lsn);
+
+    thread::spawn(move || lsn_lease_loop(configs, cmd));
+}
+
+fn lsn_lease_loop(configs: Vec<postgres::Config>, cmd: String) {
+    loop {
+        match lsn_lease_request(&configs, &cmd) {
+            Ok(valid_until) => {
+                let now = SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .expect("Time went backwards")
+                    .as_millis();
+
+                let sleep_duration_millis = valid_until - now - 60_000;
```

Comment:
If the server gives the client a validity less than 60 seconds, this loop will busy-poll.  Let's avoid that; maybe subtract 60 second by default, but never more than half the validity period?

Reply:
f335b19023f6c0e0e999c84d0523c6f51c95010f